src/connectors/impls/gbq/writer/sink.rs

```diff
@@ -0,0 +1,443 @@
+
+            TableType::String
+            // YYYY-[M]M-[D]D
+            | TableType::Date
+            // [H]H:[M]M:[S]S[.DDDDDD|.F]
+            | TableType::Time
+            // YYYY-[M]M-[D]D[( |T)[H]H:[M]M:[S]S[.F]]
+            | TableType::Datetime
+            // The GEOGRAPHY type is based on the OGC Simple Features specification (SFS)
+            | TableType::Geography
+            // String, because it's a precise, f32/f64 would lose precision
+            | TableType::Numeric
+            | TableType::Bignumeric
+            // [sign]Y-M [sign]D [sign]H:M:S[.F]
+            | TableType::Interval
+            | TableType::Json
+            // YYYY-[M]M-[D]D[( |T)[H]H:[M]M:[S]S[.F]][time zone]
+            | TableType::Timestamp => field_descriptor_proto::Type::String,
+            TableType::Struct => {
+                let type_name_for_field = format!("struct_{}", raw_field.name);
+                let mapped = map_field(&type_name_for_field, &raw_field.fields);
+                nested_types.push(mapped.0);
+                subfields = mapped.1;
+
+                type_name = Some(type_name_for_field);
+                field_descriptor_proto::Type::Message
+            }
+
+            TableType::Unspecified => {
+                warn!("Found a field of unspecified type: {}", raw_field.name);
+                continue;
+            }
+        };
+
+        proto_fields.push(FieldDescriptorProto {
+            name: Some(raw_field.name.to_string()),
+            number: Some(i32::from(tag)),
+            label: None,
+            r#type: Some(i32::from(grpc_type)),
+            type_name,
+            extendee: None,
+            default_value: None,
+            oneof_index: None,
+            json_name: None,
+            options: None,
+            proto3_optional: None,
+        });
+
+        fields.insert(
+            raw_field.name.to_string(),
+            Field {
+                table_type,
+                tag: u32::from(tag),
+                subfields,
+            },
+        );
+
+        tag += 1;
+    }
+
+    (
+        DescriptorProto {
+            name: Some(schema_name.to_string()),
+            field: proto_fields,
+            extension: vec![],
+            nested_type: nested_types,
+            enum_type: vec![],
+            extension_range: vec![],
+            oneof_decl: vec![],
+            options: None,
+            reserved_range: vec![],
+            reserved_name: vec![],
+        },
+        fields,
+    )
+}
+
+fn encode_field(val: &Value, field: &Field, result: &mut Vec<u8>) -> Result<()> {
+    let tag = field.tag;
+
+    // fixme check which fields are required and fail if they're missing
+    // fixme do not panic if the tremor type does not match
+    match field.table_type {
+        TableType::Double => prost::encoding::double::encode(
+            tag,
+            &val.as_f64().ok_or_else(|| {
+                ErrorKind::BigQueryTypeMismatch("f64", format!("{:?}", val.value_type()))
```

Comment:
Since the `BigQueryTypeMismatch`'s second argument seems to always be a value_type, would it be possible to use that as the 2nd argument for `BigQueryTypeMismatch` instead of converting it to a string each time?

Reply:
As a site note, perhaps a `try_as_*` function in the value_traits that returns the actual type as an error would be super nice I'll throw in a ticket for that in the value_trait repo